Build fails on master in Windows

**Env：win10 1909
Node.js：both v8.11.4 and v12.13.1
When I run 'npm run build'，there is a error as below:**

`
Running "build:all:*" (build) task                                                                                       Warning: "var/arr.js" does not exist in the hypothetical file system! Use --force to continue.                                                                                                                                                    Aborted due to warnings.                                                                                                 npm ERR! code ELIFECYCLE                                                                                                 npm ERR! errno 3                                                                                                         npm ERR! jquery@4.0.0-pre build: `npm install && grunt`                                                                  npm ERR! Exit status 3                                                                                                   npm ERR!                                                                                                                 npm ERR! Failed at the jquery@4.0.0-pre build script.                                                                    npm ERR! This is probably not a problem with npm. There is likely additional logging output above.                                                                                                                                                npm ERR! A complete log of this run can be found in:                                                                     npm ERR!     E:\npm_modules\npm_cache\_logs\2019-11-22T14_45_16_106Z-debug.log
--
`
**then, I viewed the log**:
`13 verbose stack Error: jquery@4.0.0-pre build: `npm install && grunt`
13 verbose stack Exit status 3
13 verbose stack     at EventEmitter.<anonymous> (D:\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:210:5)
13 verbose stack     at ChildProcess.<anonymous> (D:\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:210:5)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
14 verbose pkgid jquery@4.0.0-pre
15 verbose cwd E:\workspace\jquery
16 verbose Windows_NT 10.0.18363
17 verbose argv "D:\\nodejs\\node.exe" "D:\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
18 verbose node v12.13.1
19 verbose npm  v6.12.1
20 error code ELIFECYCLE
21 error errno 3
22 error jquery@4.0.0-pre build: `npm install && grunt`
22 error Exit status 3
23 error Failed at the jquery@4.0.0-pre build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 3, true ]
`
I cant got what I can do.

--- a/build/tasks/build.js
+++ b/build/tasks/build.js
@@ -10,7 +10,7 @@ module.exports = function( grunt ) {
 	const fs = require( "fs" );
 	const path = require( "path" );
 	const rollup = require( "rollup" );
-	const rollupHypothetical = require( "rollup-plugin-hypothetical" );
+	const rollupFileOverrides = require( "./lib/rollup-plugin-file-overrides" );
 	const Insight = require( "insight" );
 	const pkg = require( "../../package.json" );
 	const srcFolder = path.resolve( `${ __dirname }/../../src` );
@@ -39,10 +39,18 @@ module.exports = function( grunt ) {
 		outro: wrapper[ 1 ]
 			.replace( /^\n*/, "" )
 	};
-	const rollupHypotheticalOptions = {
-		allowFallthrough: true,
-		files: {}
-	};
+	const fileOverrides = new Map();
+
+	function getOverride( filePath ) {
+		return fileOverrides.get( path.resolve( filePath ) );
+	}
+
+	function setOverride( filePath, source ) {
+
+		// We want normalized paths in overrides as they will be matched
+		// against normalized paths in the file overrides Rollup plugin.
+		fileOverrides.set( path.resolve( filePath ), source );
+	}
 
 	grunt.registerMultiTask(
 		"build",
@@ -186,14 +194,14 @@ module.exports = function( grunt ) {
 
 			// Remove the jQuery export from the entry file, we'll use our own
 			// custom wrapper.
-			rollupHypotheticalOptions.files[ inputRollupOptions.input ] = read( inputFileName )
-				.replace( /\n*export default jQuery;\n*/, "\n" );
+			setOverride( inputRollupOptions.input,
+				read( inputFileName ).replace( /\n*export default jQuery;\n*/, "\n" ) );
 
 			// Replace exports/global with a noop noConflict
 			if ( ( index = excluded.indexOf( "exports/global" ) ) > -1 ) {
-				rollupHypotheticalOptions.files[ `${ srcFolder }/exports/global.js` ] =
+				setOverride( `${ srcFolder }/exports/global.js`,
 					"import jQuery from \"../core.js\";\n\n" +
-					"jQuery.noConflict = function() {};";
+						"jQuery.noConflict = function() {};" );
 				excluded.splice( index, 1 );
 			}
 
@@ -207,9 +215,10 @@ module.exports = function( grunt ) {
 				}
 
 				// Remove the comma for anonymous defines
-				rollupHypotheticalOptions.files[ `${ srcFolder }/exports/amd.js` ] =
+				setOverride( `${ srcFolder }/exports/amd.js`,
 					read( "exports/amd.js" )
-						.replace( /(\s*)"jquery"(\,\s*)/, amdName ? "$1\"" + amdName + "\"$2" : "" );
+						.replace( /(\s*)"jquery"(\,\s*)/,
+							amdName ? "$1\"" + amdName + "\"$2" : "" ) );
 			}
 
 			grunt.verbose.writeflags( excluded, "Excluded" );
@@ -225,7 +234,7 @@ module.exports = function( grunt ) {
 
 				// Replace excluded modules with empty sources.
 				for ( const module of excluded ) {
-					rollupHypotheticalOptions.files[ `${ srcFolder }/${ module }.js` ] = "";
+					setOverride( `${ srcFolder }/${ module }.js`, "" );
 				}
 			}
 
@@ -234,20 +243,21 @@ module.exports = function( grunt ) {
 
 				// Remove the default inclusions, they will be overwritten with the explicitly
 				// included ones.
-				rollupHypotheticalOptions.files[ inputRollupOptions.input ] = "";
+				setOverride( inputRollupOptions.input, "" );
 
 			}
 
 			// Import the explicitly included modules.
 			if ( included.length ) {
-				rollupHypotheticalOptions.files[ inputRollupOptions.input ] += included
-					.map( module => `import "./${module}.js";` )
-					.join( "\n" );
+				setOverride( inputRollupOptions.input,
+					getOverride( inputRollupOptions.input ) + included
+						.map( module => `import "./${module}.js";` )
+						.join( "\n" ) );
 			}
 
 			const bundle = await rollup.rollup( {
 				...inputRollupOptions,
-				plugins: [ rollupHypothetical( rollupHypotheticalOptions ) ]
+				plugins: [ rollupFileOverrides( fileOverrides ) ]
 			} );
 
 			const { output: [ { code } ] } = await bundle.generate( outputRollupOptions );

--- a/build/tasks/lib/rollup-plugin-file-overrides.js
+++ b/build/tasks/lib/rollup-plugin-file-overrides.js
@@ -1,0 +1,24 @@
+"use strict";
+
+/**
+ * A Rollup plugin accepting a file overrides map and changing
+ * module sources to the overridden ones where provided. Files
+ * without overrides are loaded from disk.
+ *
+ * @param {Map<string, string>} fileOverrides
+ */
+module.exports = ( fileOverrides ) => {
+	return {
+		name: "jquery-file-overrides",
+		load( id ) {
+			if ( fileOverrides.has( id ) ) {
+
+				// Replace the module by a fake source.
+				return fileOverrides.get( id );
+			}
+
+			// Handle this module via the file system.
+			return null;
+		}
+	};
+};

--- a/package.json
+++ b/package.json
@@ -61,7 +61,6 @@
     "raw-body": "2.3.3",
     "requirejs": "2.3.6",
     "rollup": "1.25.2",
-    "rollup-plugin-hypothetical": "2.1.0",
     "sinon": "7.3.1",
     "strip-json-comments": "2.0.1",
     "testswarm": "1.1.0",

